webapp/channels/src/components/suggestion/modal_suggestion_list.tsx

```diff
@@ -72,7 +72,8 @@ export default class ModalSuggestionList extends React.PureComponent<Props, Stat
     calculateInputRect = () => {
         if (this.props.inputRef.current) {
             const rect = this.props.inputRef.current.getBoundingClientRect();
-            return {top: rect.top, bottom: rect.bottom, width: rect.width};
+            const rectHeight = rect.bottom - rect.top;
+            return {top: rect.top - (2.1 * rectHeight), bottom: rect.bottom + (1.8 * rectHeight), width: rect.width};
```

Comment:
Where do the numbers 2.1 and 1.8 numbers come from here? 

I'm trying to understand how this math works, and I wouldn't expect us that subtracting something from the `top` value would move the suggestion list downwards

Reply:
I am a bit discontent with that 2.1 and 1.8 is chosen by trial and error. It seemed both top and bottom position was incorrectly chosen before. If there is a better way of doing this I would favor that. Do you have any suggestion how that could be done or does it suffice to do it as such?

Top was before placed too far down and thus if you were to switch from `bottom` to `top` on `listPosition='bottom'` it would before also be placed incorrectly. Now that should be a viable option where the dropdown suggestion is position correctly.

Reply:
I'm not sure if `this.state.position` or `this.props.position` are actually the issue here. I'm testing this locally, and they're always set to `bottom` when I have the app maximized.

Since it's always set to be on the bottom, I think the issue might be with how we're calculating the `top` position of the list in the render function. Since the list has fixed positioning, I'm not sure why the top of the list isn't always set to the bottom edge of the input, so maybe there's an issue there

Reply:
I've noticed that, if we don't set the `top` style of the list in the render method, the list seems to be positioned correctly, but that's different than setting it to `0`. I didn't expect that since it has `position: fixed`

Reply:
From my testing `this.state.position` and `this.props.position` are always the same as what is defined in `<SuggestionBox ... >` and `top` on default, how do you make it to always be `bottom`?. 

I managed to get it placed correctly as you mention by skipping `{top: ...}` in the `<SuggestionList ...>`. But that is only true with `bottom`, if we switch to `top`, we get it exactly over (Which is expected according to how it behaves with `bottom`) the input field. 

Do you think we instead should change how `bottom` is defined when picking `top` with `position = {bottom: ...}` and simply not have the `else { position = ... }` at row 204 in `modal_suggestion_box.ts` and thus when picking `bottom` we ignore any position arguments?

Thanks for the feedback!

Reply:
I don't think I'm doing anything special. I added a log statement to the render method that prints `this.state.position` and it's always `bottom` for that modal.

<img width="1840" alt="Screenshot 2023-11-16 at 4 56 07 PM" src="https://github.com/mattermost/mattermost/assets/3277310/33e39e64-1a17-437a-a283-cba74ce8d9c9">


> I managed to get it placed correctly as you mention by skipping `{top: ...}` in the `<SuggestionList ...>`. But that is only true with `bottom`, if we switch to `top`, we get it exactly over (Which is expected according to how it behaves with `bottom`) the input field.
> 
> Do you think we instead should change how `bottom` is defined when picking `top` with `position = {bottom: ...}` and simply not have the `else { position = ... }` at row 204 in `modal_suggestion_box.ts` and thus when picking `bottom` we ignore any position arguments?

If that lines up the box when it opens downwards, that would work for me. It's definitely possible that we used to need to specify a `top` value but some other CSS changed so that we don't need it here

Reply:
Alright! I made the change I suggested and changed the preview image in the PR for you to see the result. Thanks for the help.

Reply:
Thanks for the change! This works great now, and it gets rid of the small gap that the previous version of this PR had.

Do we still need this change to how we calculate `top` and `bottom` then? I tried removing it locally, and the dropdown is still correctly positioned when the dropdown opens downwards